multiqc/modules/dragen/coverage_metrics.py

```diff
@@ -1,346 +1,381 @@
-
-        if not data_by_sample:
-            return set()
-
-        # Write data to file
-        self.write_data_file(data_by_sample, "dragen_cov_metrics")
-
-        all_metric_names = set()
-        for sn, sdata in data_by_sample.items():
-            for m in sdata.keys():
-                all_metric_names.add(m)
-        gen_stats_headers, own_tabl_headers = make_headers(all_metric_names, WGS_COV_METRICS)
-
-        self.general_stats_addcols(data_by_sample, gen_stats_headers, namespace=NAMESPACE)
-
-        return self._create_table(
-            data_by_sample,
-            own_tabl_headers,
-            "WGS Coverage Metrics",
-            "dragen-wgs-coverage-metrics",
-        )
+        if not cov_data: return set()
+
+        '''""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+        Coverage data is collected now. The headers are also complete and available for use.
+        If other files (eg json) are needed, then these can be found with find_log_files,
+        processed and used afterwards. See afterqc.py or supernova.py for example.
+        -------------------------------------------------------------------------------------
+        CSS/JS can be imported, see fastqc.py for details.
+        -------------------------------------------------------------------------------------
+        The ignore_samples and clean_s_name use the Multiqc's config module.
+
+        Assumption: some/most users are not or dont want to be familiar with the cleaning
+        details of the code and most will set configurations for the original sample names.
+        If it is true, then the proper sequence of execution steps would be:
+        1. Ignore samples.
+        2. Use the global clean_s_name.
+        3. Use the local dragen cleaner.
+
+        If not, then the sequence can be always rearranged.
+        --------------------------------------------------------------------------------------
+        In theory some combinations of sample names, configurations and implementation aspects
+        of both cleaners may produce duplicates. Cleaned names are checked for duplicates, if
+        presented then their original names are used instead. The issue will be reported.
+        """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""'''
+        cov_data = self.ignore_samples(cov_data)
+        check_cleaned_names(
+        {s_name: clean_sample(self.clean_s_name(s_name)) for s_name in cov_data.keys()})
 
-    def _create_table(self, data_by_sample, own_tabl_headers, table_name: str, table_anchor: str):
+        # Write data into the general table.
+        gen_data, gen_headers = make_general_stats()
+        self.general_stats_addcols(gen_data, gen_headers, namespace = NAMESPACE)
+
+        # Write general data to file.
+        self.write_data_file(gen_data, "dragen_cov_metrics")
+
+        # Report all found errors and warnings.
+        make_log_report()
```

Comment:
The whole log reporting is just an unnecessary addition. It does not play any significant role in the code. 
I just wanted to make it sort of a "background function", which is good to be aware of, but it does not really matter how it works. I'll adjust it.

Reply:
I can understand that, from a viewers perspective reducing the number of "side-effects" is helpful, as otherwise it is difficult to discern which objects you are creating a logging report on.

Reply:
I'll add an argument for log data then for now.